### PR TITLE
GitHub workflows for deployment preview

### LINKED ISSUE #183: Add Github workflow for deployment preview
#### Current Behavior
Currently, we don't have any GitHub workflow defined for deployment preview of `Sistent` site.
Check this PR -> https://github.com/layer5io/sistent/pull/182

#### Desired Behavior

-> Add  GitHub workflow for deployment preview.

#### Mockups

---

**[Contributor Guide](https://docs.meshery.io/project/contributing)**

- 🎨 Wireframes and [designs for Layer5 site](https://www.figma.com/file/5ZwEkSJwUPitURD59YHMEN/Layer5-Designs) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)

### PR DESCRIPTION
**Pull request by coder12git**
**Notes for Reviewers**

This PR fixes #183

**[Signed commits](../blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**

- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by coder12git** [comment]
@leecalcote i have made some changes here, assuming that we have a `site` folder in `master` branch, and those deployments workflow runs inside that `site `folder, because this is the folder in my `sistent-site-draft ` PR which contains the gatsby site

**Comment by coder12git** [comment]
> There is a syntax error in the github actions workflow: see link: https://github.com/layer5io/sistent/actions/runs/6754135751/workflow

fixed

**Review by leecalcote — APPROVED** [review]
Any deviation from layer5io/layer5?

File: .github/workflows/build-and-deploy-site.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+          cd site
+          npm install
+          npm run build
+
+      - name: Deploy 🚀
+        uses: JamesIves/github-pages-deploy-action@4.0.0
+        with:
+          branch: master # The branch the action should deploy to.
+          folder: site/public # The folder the action should deploy.
+          clean: true
```
Comment (leecalcote) [inline .github/workflows/build-and-deploy-site.yml:28]:
```suggestion
          branch: site # The branch the action should deploy to.
          folder: public # The folder the action should deploy.
          clean: true
```

Comment (leecalcote) [reply .github/workflows/build-and-deploy-site.yml:28]: Aligning with existing practices.

**Pull request merged**